segfault on setup

### Summary

after following gopass/docs/setup.md I run gopass and choose "Y":

```
It seems you are new to gopass. Do you want to run the onboarding wizard? [Y/n/q]: Y
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x43110c]

goroutine 1 [running]:
github.com/gopasspw/gopass/internal/action.(*Action).InitOnboarding(0x40001e7420, 0x400008f000, 0x7b2fd4, 0x49)
        /home/stefan/go/src/github.com/gopasspw/gopass/internal/action/init.go:209 +0x3e4
github.com/gopasspw/gopass/internal/action.(*Action).Initialized(0x40001e7420, 0x400008f000, 0x20, 0x643a00)
        /home/stefan/go/src/github.com/gopasspw/gopass/internal/action/init.go:42 +0x1d4
main.setupApp.func2(0x400008f000, 0x40000afa00, 0x6)
        /home/stefan/go/src/github.com/gopasspw/gopass/main.go:121 +0x34
github.com/urfave/cli.(*App).RunContext(0x4000081200, 0x89ff00, 0x4000146fc0, 0x400009c1c0, 0x1, 0x1, 0x0, 0x0)
        /home/stefan/go/src/github.com/urfave/cli/app.go:315 +0x514
main.main()
        /home/stefan/go/src/github.com/gopasspw/gopass/main.go:81 +0x310
```

### Steps To Reproduce

Do the same, but perhaps it's architecture-dependend, since this is ubuntu 20.4 in WSL version 1 on Windows 10, arm64. Here the output of `uname -a`:

```
Linux glarm64dos 4.4.0-18362-Microsoft #836-Microsoft Mon May 05 16:04:00 PST 2020 aarch64 aarch64 aarch64 GNU/Linux
```

### Expected behavior
 
normal initialisation-process

### Environment

- OS: Ubuntu 20.04
- OS version: Linux glarm64dos 4.4.0-18362-Microsoft #836-Microsoft Mon May 05 16:04:00 PST 2020 aarch64 aarch64 aarch64 GNU/Linux
- gopass Version: no output, since it tries to initialise instead of outputting the version.
- Installation method: from source: `go get -u github.com/gopasspw/gopass`

### Additional context

It worked in WSL 2, but Microsoft downgraded me, since there seems to be a bug in the Hyper-V code on arm64.

--- a/internal/action/clone.go
+++ b/internal/action/clone.go
@@ -133,6 +133,7 @@ func (s *Action) cloneGetGitConfig(ctx context.Context, name string) (string, st
 	// NB: discarding returned error since this is merely a best-effort look-up for convenience
 	username, email, _ := cui.AskForGitConfigUser(ctx, s.Store.Crypto(ctx, name))
 	if username == "" {
+		username = termio.DetectName(ctx, nil)
 		var err error
 		username, err = termio.AskForString(ctx, color.CyanString("Please enter a user name for password store git config"), username)
 		if err != nil {
@@ -140,6 +141,7 @@ func (s *Action) cloneGetGitConfig(ctx context.Context, name string) (string, st
 		}
 	}
 	if email == "" {
+		email = termio.DetectEmail(ctx, nil)
 		var err error
 		email, err = termio.AskForString(ctx, color.CyanString("Please enter an email address for password store git config"), email)
 		if err != nil {

--- a/internal/action/clone_test.go
+++ b/internal/action/clone_test.go
@@ -13,6 +13,7 @@ import (
 	"github.com/gopasspw/gopass/internal/config"
 	"github.com/gopasspw/gopass/internal/gptest"
 	"github.com/gopasspw/gopass/internal/out"
+	"github.com/gopasspw/gopass/internal/termio"
 	"github.com/gopasspw/gopass/pkg/ctxutil"
 
 	"github.com/blang/semver"
@@ -112,6 +113,11 @@ func TestCloneBackendIsStoredForMount(t *testing.T) {
 func TestCloneGetGitConfig(t *testing.T) {
 	u := gptest.NewUnitTester(t)
 	defer u.Remove()
+
+	r1 := gptest.UnsetVars(termio.NameVars)
+	defer r1()
+	r2 := gptest.UnsetVars(termio.EmailVars)
+	defer r2()
 
 	ctx := context.Background()
 	ctx = ctxutil.WithAlwaysYes(ctx, true)

--- a/internal/action/commands.go
+++ b/internal/action/commands.go
@@ -413,15 +413,18 @@ func (s *Action) GetCommands() []*cli.Command {
 				},
 				&cli.StringFlag{
 					Name:  "crypto",
-					Usage: "Select crypto backend (gpg, gpgcli, plain, xc)",
+					Usage: "Select crypto backend (gpgcli, plain, xc)",
+					Value: "gpgcli",
 				},
 				&cli.StringFlag{
 					Name:  "rcs",
 					Usage: "Select sync backend (git, gitcli, noop)",
+					Value: "gitcli",
 				},
 				&cli.StringFlag{
 					Name:  "storage",
 					Usage: "Select storage backend (fs, inmen, ondisk)",
+					Value: "fs",
 				},
 			},
 		},

--- a/internal/action/git.go
+++ b/internal/action/git.go
@@ -6,6 +6,7 @@ import (
 
 	"github.com/gopasspw/gopass/internal/backend"
 	"github.com/gopasspw/gopass/internal/cui"
+	"github.com/gopasspw/gopass/internal/debug"
 	"github.com/gopasspw/gopass/internal/out"
 	"github.com/gopasspw/gopass/internal/termio"
 	"github.com/gopasspw/gopass/pkg/ctxutil"
@@ -36,7 +37,7 @@ func (s *Action) GitInit(c *cli.Context) error {
 
 func (s *Action) rcsInit(ctx context.Context, store, un, ue string) error {
 	bn := backend.RCSBackendName(backend.GetRCSBackend(ctx))
-	out.Green(ctx, "Initializing git repository (%s) ...", bn)
+	out.Green(ctx, "Initializing git repository (%s) for %s / %s...", bn, un, ue)
 
 	userName, userEmail := s.getUserData(ctx, store, un, ue)
 	if err := s.Store.GitInit(ctx, store, userName, userEmail); err != nil {
@@ -50,31 +51,39 @@ func (s *Action) rcsInit(ctx context.Context, store, un, ue string) error {
 	return nil
 }
 
-func (s *Action) getUserData(ctx context.Context, store, un, ue string) (string, string) {
-	if un != "" && ue != "" {
-		return un, ue
+func (s *Action) getUserData(ctx context.Context, store, name, email string) (string, string) {
+	if name != "" && email != "" {
+		debug.Log("Username: %s, Email: %s (provided)", name, email)
+		return name, email
 	}
 
 	// for convenience, set defaults to user-selected values from available private keys
 	// NB: discarding returned error since this is merely a best-effort look-up for convenience
 	userName, userEmail, _ := cui.AskForGitConfigUser(ctx, s.Store.Crypto(ctx, store))
 
-	if userName == "" {
+	if name == "" {
+		if userName == "" {
+			userName = termio.DetectName(ctx, nil)
+		}
 		var err error
-		userName, err = termio.AskForString(ctx, color.CyanString("Please enter a user name for password store git config"), userName)
+		name, err = termio.AskForString(ctx, color.CyanString("Please enter a user name for password store git config"), userName)
 		if err != nil {
 			out.Error(ctx, "Failed to ask for user input: %s", err)
 		}
 	}
-	if userEmail == "" {
+	if email == "" {
+		if userEmail == "" {
+			userEmail = termio.DetectEmail(ctx, nil)
+		}
 		var err error
-		userEmail, err = termio.AskForString(ctx, color.CyanString("Please enter an email address for password store git config"), userEmail)
+		email, err = termio.AskForString(ctx, color.CyanString("Please enter an email address for password store git config"), userEmail)
 		if err != nil {
 			out.Error(ctx, "Failed to ask for user input: %s", err)
 		}
 	}
 
-	return userName, userEmail
+	debug.Log("Username: %s, Email: %s (detected)", name, email)
+	return name, email
 }
 
 // GitAddRemote adds a new git remote

--- a/internal/action/history_test.go
+++ b/internal/action/history_test.go
@@ -11,6 +11,7 @@ import (
 	"github.com/gopasspw/gopass/internal/config"
 	"github.com/gopasspw/gopass/internal/gptest"
 	"github.com/gopasspw/gopass/internal/out"
+	"github.com/gopasspw/gopass/internal/termio"
 	"github.com/gopasspw/gopass/pkg/ctxutil"
 
 	"github.com/blang/semver"
@@ -22,6 +23,11 @@ import (
 func TestHistory(t *testing.T) {
 	u := gptest.NewUnitTester(t)
 	defer u.Remove()
+
+	r1 := gptest.UnsetVars(termio.NameVars)
+	r2 := gptest.UnsetVars(termio.EmailVars)
+	defer r1()
+	defer r2()
 
 	ctx := context.Background()
 	ctx = ctxutil.WithAlwaysYes(ctx, true)
@@ -49,6 +55,7 @@ func TestHistory(t *testing.T) {
 
 	// init git
 	require.NoError(t, act.rcsInit(ctx, "", "foo bar", "foo.bar@example.org"))
+	t.Logf("init git: %s", buf.String())
 	buf.Reset()
 
 	// insert bar

--- a/internal/action/init.go
+++ b/internal/action/init.go
@@ -11,6 +11,7 @@ import (
 	"github.com/gopasspw/gopass/internal/cui"
 	"github.com/gopasspw/gopass/internal/debug"
 	"github.com/gopasspw/gopass/internal/out"
+	"github.com/gopasspw/gopass/internal/store/root"
 	"github.com/gopasspw/gopass/internal/termio"
 	"github.com/gopasspw/gopass/pkg/ctxutil"
 	"github.com/gopasspw/gopass/pkg/pwgen/xkcdgen"
@@ -183,29 +184,33 @@ func (s *Action) InitOnboarding(c *cli.Context) error {
 	remote := c.String("remote")
 	team := c.String("alias")
 	create := c.Bool("create")
-	name := termio.DetectName(c.Context, c)
-	if name != "" {
+
+	ctx = initParseContext(ctx, c)
+	if name := termio.DetectName(c.Context, c); name != "" {
 		ctx = ctxutil.WithUsername(ctx, name)
 	}
-	email := termio.DetectEmail(c.Context, c)
-	if email != "" {
+	if email := termio.DetectEmail(c.Context, c); email != "" {
 		ctx = ctxutil.WithEmail(ctx, email)
 	}
-	ctx = backend.WithCryptoBackendString(ctx, c.String("crypto"))
-
-	// default to git
-	if rcs := c.String("rcs"); rcs != "" {
-		ctx = backend.WithRCSBackendString(ctx, c.String("rcs"))
-	} else {
-		debug.Log("Using default RCS backend (GitCLI)")
-		ctx = backend.WithRCSBackend(ctx, backend.GitCLI)
+	// need to re-initialize the root store or it's already initialized
+	// and won't properly set up crypto according to our context.
+	s.Store = root.New(s.cfg)
+	inited, err := s.Store.Initialized(ctx)
+	if err != nil {
+		return ExitError(ExitUnknown, err, "Failed to initialized store: %s", err)
+	}
+	if inited {
+		out.Error(ctx, "WARNING: Store is already initialized")
+		return nil
 	}
 
 	ctx = out.AddPrefix(ctx, "[init] ")
-	debug.Log("Starting Onboarding Wizard - remote: %s - team: %s - create: %t - name: %s - email: %s", remote, team, create, name, email)
+	debug.Log("Starting Onboarding Wizard - remote: %s - team: %s - create: %t - name: %s - email: %s", remote, team, create, ctxutil.GetUsername(ctx), ctxutil.GetEmail(ctx))
 
-	crypto := s.getCryptoFor(ctx, name)
-
+	crypto := s.getCryptoFor(ctx, team)
+	if crypto == nil {
+		return fmt.Errorf("can not continue without crypto")
+	}
 	debug.Log("Crypto Backend initialized as: %s", crypto.Name())
 
 	// check for existing GPG keypairs (private/secret keys). We need at least
@@ -214,7 +219,7 @@ func (s *Action) InitOnboarding(c *cli.Context) error {
 		out.Yellow(ctx, "No useable crypto keys. Generating new key pair")
 		ctx := out.AddPrefix(ctx, "[crypto] ")
 		out.Print(ctx, "Key generation may take up to a few minutes")
-		if err := s.initGenerateIdentity(ctx, crypto, name, email); err != nil {
+		if err := s.initGenerateIdentity(ctx, crypto, ctxutil.GetUsername(ctx), ctxutil.GetEmail(ctx)); err != nil {
 			return errors.Wrapf(err, "failed to create new private key")
 		}
 	}

--- a/internal/action/init_test.go
+++ b/internal/action/init_test.go
@@ -25,6 +25,9 @@ func TestInit(t *testing.T) {
 	ctx := context.Background()
 	ctx = ctxutil.WithAlwaysYes(ctx, true)
 	ctx = ctxutil.WithInteractive(ctx, false)
+	ctx = backend.WithRCSBackend(ctx, backend.Noop)
+	ctx = backend.WithCryptoBackend(ctx, backend.Plain)
+	ctx = backend.WithStorageBackend(ctx, backend.FS)
 
 	act, err := newMock(ctx, u)
 	require.NoError(t, err)
@@ -41,9 +44,11 @@ func TestInit(t *testing.T) {
 	c := gptest.CliCtx(ctx, t, "foo.bar@example.org")
 	assert.NoError(t, act.Initialized(c))
 	assert.Error(t, act.Init(c))
-	assert.Error(t, act.InitOnboarding(c))
+	assert.NoError(t, act.InitOnboarding(c))
 
 	crypto := act.Store.Crypto(ctx, "")
+	require.NotNil(t, crypto)
+	assert.Equal(t, "plain", crypto.Name())
 	assert.Equal(t, true, act.initHasUseablePrivateKeys(ctx, crypto))
 	assert.Error(t, act.initGenerateIdentity(ctx, crypto, "foo bar", "foo.bar@example.org"))
 	buf.Reset()

--- a/internal/action/show.go
+++ b/internal/action/show.go
@@ -120,7 +120,7 @@ func (s *Action) showHandleOutput(ctx context.Context, name string, sec store.Se
 	}
 
 	if pw == "" && body == "" {
-		return ExitError(ExitNotFound, store.ErrNoBody, store.ErrNoBody.Error())
+		return ExitError(ExitNotFound, store.ErrEmptySecret, store.ErrEmptySecret.Error())
 	}
 
 	if IsPrintQR(ctx) && pw != "" {

--- a/internal/backend/storage/fs/store.go
+++ b/internal/backend/storage/fs/store.go
@@ -106,8 +106,9 @@ func (s *Store) Exists(ctx context.Context, name string) bool {
 		name = filepath.FromSlash(name)
 	}
 	path := filepath.Join(s.path, filepath.Clean(name))
-	debug.Log("%s exists at %s", name, path)
-	return fsutil.IsFile(path)
+	found := fsutil.IsFile(path)
+	debug.Log("Checking if %s exists at %s: %t", name, path, found)
+	return found
 }
 
 // List returns a list of all entities

--- a/internal/gptest/utils.go
+++ b/internal/gptest/utils.go
@@ -78,3 +78,17 @@ func flagset(t *testing.T, flags map[string]string, args []string) *flag.FlagSet
 
 	return fs
 }
+
+// UnsetVars will unset the specified env vars and return a restore func
+func UnsetVars(ls []string) func() {
+	old := make(map[string]string, len(ls))
+	for _, k := range ls {
+		old[k] = os.Getenv(k)
+		os.Unsetenv(k)
+	}
+	return func() {
+		for k, v := range old {
+			os.Setenv(k, v)
+		}
+	}
+}

--- a/internal/store/err.go
+++ b/internal/store/err.go
@@ -21,6 +21,8 @@ var (
 	ErrGitNoRemote = errors.Errorf("git has no remote origin")
 	// ErrGitNothingToCommit is returned if there are no staged changes
 	ErrGitNothingToCommit = errors.Errorf("git has nothing to commit")
+	// ErrEmptySecret is returned if a secret exists but has no content
+	ErrEmptySecret = errors.Errorf("empty secret")
 	// ErrNoBody is returned if a secret exists but has no content beyond a password
 	ErrNoBody = errors.Errorf("no safe content to display, you can force display with -f")
 	// ErrNoPassword is returned is a secret exists but has no password, only a body

--- a/internal/termio/identity.go
+++ b/internal/termio/identity.go
@@ -8,6 +8,21 @@ import (
 	"github.com/urfave/cli/v2"
 )
 
+var (
+	// NameVars are the env vars checked for a valid name
+	NameVars = []string{
+		"GIT_AUTHOR_NAME",
+		"DEBFULLNAME",
+		"USER",
+	}
+	// EmailVars are the env vars checked for a valid email
+	EmailVars = []string{
+		"GIT_AUTHOR_EMAIL",
+		"DEBEMAIL",
+		"EMAIL",
+	}
+)
+
 // DetectName tries to guess the name of the logged in user
 func DetectName(ctx context.Context, c *cli.Context) string {
 	cand := make([]string, 0, 5)
@@ -15,11 +30,9 @@ func DetectName(ctx context.Context, c *cli.Context) string {
 	if c != nil {
 		cand = append(cand, c.String("name"))
 	}
-	cand = append(cand,
-		os.Getenv("GIT_AUTHOR_NAME"),
-		os.Getenv("DEBFULLNAME"),
-		os.Getenv("USER"),
-	)
+	for _, k := range NameVars {
+		cand = append(cand, os.Getenv(k))
+	}
 	for _, e := range cand {
 		if e != "" {
 			return e
@@ -35,11 +48,9 @@ func DetectEmail(ctx context.Context, c *cli.Context) string {
 	if c != nil {
 		cand = append(cand, c.String("email"))
 	}
-	cand = append(cand,
-		os.Getenv("GIT_AUTHOR_EMAIL"),
-		os.Getenv("DEBEMAIL"),
-		os.Getenv("EMAIL"),
-	)
+	for _, k := range EmailVars {
+		cand = append(cand, os.Getenv(k))
+	}
 	for _, e := range cand {
 		if e != "" {
 			return e

--- a/main.go
+++ b/main.go
@@ -134,6 +134,11 @@ func setupApp(ctx context.Context, sv semver.Version) (context.Context, *cli.App
 			Usage: "Assume yes on all yes/no questions or use the default on all others",
 		},
 		&cli.BoolFlag{
+			Name:    "force",
+			Aliases: []string{"f"},
+			Usage:   "Force displaying content",
+		},
+		&cli.BoolFlag{
 			Name:    "clip",
 			Aliases: []string{"c"},
 			Usage:   "Copy the first line of the secret into the clipboard",

--- a/main_test.go
+++ b/main_test.go
@@ -77,7 +77,6 @@ var commandsWithError = map[string]struct{}{
 	".otp":               {},
 	".recipients.add":    {},
 	".recipients.remove": {},
-	".setup":             {},
 	".show":              {},
 	".templates.edit":    {},
 	".templates.remove":  {},

